sdk/tests/conformance2/textures/misc/tex-3d-size-limit.html

```diff
@@ -80,6 +73,39 @@
         wtu.glErrorShouldBe(gl, gl.INVALID_VALUE, "texImage3D should fail for dimension out of range.");
     }
 
+    // Probe to discover the max non-OOM level.
+    var maxTestLevels = maxLevels;
+    while (true) {
+        gl.texImage3D(gl.TEXTURE_3D, maxTestLevels-1, gl.RGBA, 1, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+        var err = gl.getError();
+        if (err == gl.OUT_OF_MEMORY) {
+            debug("Probe failed for level=" + (maxTestLevels-1) + ", reducing...");
+            maxTestLevels -= 1;
+            if (!maxTestLevels) {
+                testFailed("Failed to allocate any levels");
+                return;
+            }
+            continue;
+        }
+        if (err) {
+            testFailed("Should not hit non-OOM errors during max level probing.");
+            return;
+        }
+        break;
+    }
+
+    for (var i = 0; i < maxTestLevels; ++i) {
+        var level = maxTestLevels - i - 1;
+        var size = 1 << i;
+
+        gl.texImage3D(gl.TEXTURE_3D, level, gl.RGBA, size, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
+        gl.texImage3D(gl.TEXTURE_3D, level, gl.RGBA, 1, size, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
+        gl.texImage3D(gl.TEXTURE_3D, level, gl.RGBA, 1, 1, size, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
```

Comment:
being nit-pikey : again is there no way to enumerate these test conditions in JS? rather than having so many asserts that imply the same message and are not very explicit on where and what failed?  it seems that it just returns 3 pass messages rather than what component passed exactly.

Reply:
It's a fair point that these error messages could read something like "texImage3D with maximum [x/y/z] dimension should succeed".

Reply:
Got it :D Thanks.